src/Simple/JSON.purs

```diff
@@ -221,6 +222,15 @@ instance readFieldsCons ::
       name = reflectSymbol nameP
       withExcept' = withExcept <<< map $ ErrorAtProperty name
 
+exceptTApply :: forall a b e m. Semigroup e => Applicative m => ExceptT e m (a -> b) -> ExceptT e m a -> ExceptT e m b
+exceptTApply fun a = ExceptT $ ado
+  fun' :: Either e (a -> b) <- runExceptT fun
+  a' :: (Either e a) <- runExceptT a
+  in toEither (apply (fromEither fun') (fromEither a'))
```

Comment:
is it actually necessary to use Validation? if you're going to runExceptT, there's already a price paid for the conversions, so at this point it seems what should be done here is to pattern match on the four cases of LL,LR,RL,RR